helix-core/src/main/java/org/apache/helix/controller/stages/CustomizedViewAggregationStage.java

```diff
@@ -152,6 +163,33 @@ private void computeCustomizedStateView(final Resource resource, final String st
     if (curCustomizedView == null || !curCustomizedView.getRecord().equals(view.getRecord())) {
       // Add customized view to the list which will be written to ZK later.
       updatedCustomizedViews.add(view);
+      updatedStartTimestamps.put(resourceName,
+          customizedStateOutput.getResourceStartTimeMap(stateType, resourceName));
     }
   }
+
+  private CustomizedViewMonitor getOrCreateMonitor(ClusterEvent event) throws JMException {
```

Comment:
Once we put this map into the cache object, concurrent control is necessary for this method.